### PR TITLE
Publish Metrics from KEA containers

### PR DESCRIPTION
**Pull request by emileswarts**
We will take stats from the KEA API and push them to Cloudwatch metrics.
This will be used in Grafana.

### CODE REVIEW

File: dhcp-service/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,12 @@
 FROM alpine:3.11
 
 ENV GLIBC_VER=2.31-r0
+ARG LOCAL_DEVELOPMENT
 
 RUN apk add bash curl mysql mysql-client && \
     curl -1sLf 'https://dl.cloudsmith.io/public/isc/kea-1-8/cfg/setup/bash.alpine.sh' |  bash && \
     apk upgrade && \
-    apk add isc-kea-admin isc-kea-perfdhcp isc-kea-dhcp4 isc-kea-ctrl-agent isc-kea-hook-lease-cmds isc-kea-hook-stat-cmds && \
+    apk add ruby ruby-json ruby-dev isc-kea-admin isc-kea-perfdhcp isc-kea-dhcp4 isc-kea-ctrl-agent isc-kea-hook-lease-cmds isc-kea-hook-stat-cmds && \
```
Comment (jbevan4) [inline dhcp-service/Dockerfile:9]: How much space has this added to the image? Is it significant enough to notice a difference?

Comment (emileswarts) [reply dhcp-service/Dockerfile:9]: Looks like it's grown by about 20Mb.
Going to pull out the AWS CLI (for manual failure sns notifications), which should reclaim some space in the future.

File: dhcp-service/Dockerfile [hunk 2]
```diff
@@ -17,14 +18,20 @@ RUN apk add bash curl mysql mysql-client && \
...
 COPY ./config.json ./test_config.json
+COPY ./metrics ./metrics
 COPY ./control-agent-config.json /etc/kea/control-agent-config.json
 
 EXPOSE 67/udp
 EXPOSE 68/udp
 EXPOSE 8000
 
+RUN if [ ${LOCAL_DEVELOPMENT} == 'true' ]; then \
+  gem install rspec; \
```
Comment (jivdhaliwal) [inline dhcp-service/Dockerfile:34]: Do we see ourselves adding more gems in the future? And should we consider using bundler to manage versions / dependencies or is it ok to just do a gem install for now?

**Review by neilkidd — APPROVED** [review]

File: dhcp-service/Dockerfile (hunk 1 above)
Comment (jivdhaliwal) [inline dhcp-service/Dockerfile:9]: Do we want to lock down the ruby version and rspec version? Or just grab the latest and let the `gem install rspec` grab the latest supported version?

File: Makefile [hunk 3]
```diff
@@ -27,6 +27,10 @@ stop:
 run: start-db
 	$(DOCKER_COMPOSE) up -d dhcp
 
+test-metrics: run build-dev
```
Comment (jivdhaliwal) [inline Makefile:30]: Could we merge this with the other testing task so they both run when you do `make test` ? Or do they need to be separate?

Comment (emileswarts) [reply Makefile:30]: Makes sense yeah 👍

File: dhcp-service/Dockerfile (hunk 2 above)
Comment (emileswarts) [reply dhcp-service/Dockerfile]: Don't see us adding many more gems in the future.
Locking down versions still seems like a good idea.
Switched to using Bundler

File: dhcp-service/Dockerfile (hunk 1 above)
Comment (emileswarts) [reply dhcp-service/Dockerfile]: Locked down this ruby version

**Review by jivdhaliwal — APPROVED** [review]
Nice 👍  :shipit:

**Pull request merged**